macros/src/actix.rs

```diff
@@ -845,20 +845,27 @@ fn handle_field_struct(
             field_name = prop.rename(&field_name);
         }
 
+        let flatten = SerdeFlatten::from_field_attrs(&field.attrs);
+
         let (ty_ref, is_required) = get_field_type(&field);
 
         let docs = extract_documentation(&field.attrs);
         let docs = docs.trim();
 
-        let mut gen = quote!(
-            {
+        let mut gen = if !flatten {
```

Comment:
Once we rename the function (as mentioned below), this could be `if SerdeFlatten::exists(&field.attrs) { ... }`

Reply:
Cool, this looks better now. Just applied all suggestions